cmd/cofidectl/cmd/clusters/clusters.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2024 Cofide Limited.
+// SPDX-License-Identifier: Apache-2.0
+
+package clusters
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	cmdcontext "github.com/cofide/cofidectl/pkg/cmd/context"
+	"github.com/olekukonko/tablewriter"
+	"github.com/spf13/cobra"
+)
+
+var clustersListCmdDesc = `
+This command consists of multiple sub-commands to interact with clusters
+`
+
+type ClustersCommand struct {
+	cmdCtx *cmdcontext.CommandContext
+}
+
+func NewClustersCommand(cmdCtx *cmdcontext.CommandContext) *ClustersCommand {
+	return &ClustersCommand{
+		cmdCtx: cmdCtx,
+	}
+}
+
+func (c *ClustersCommand) GetRootCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "clusters",
+		Short: "Manage clusters",
+		Long:  clustersListCmdDesc,
+	}
+
+	cmd.AddCommand(c.getListClustersCommand())
+
+	return cmd
+}
+
+func (c *ClustersCommand) getListClustersCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "list",
+		Short: "List clusters",
+		Long:  clustersListCmdDesc,
+		Args:  cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return c.ListClusters(cmd.Context())
+		},
+	}
+
+	return cmd
+}
+
+func (c *ClustersCommand) ListClusters(ctx context.Context) error {
+	ds, err := c.cmdCtx.PluginManager.GetDataSource(ctx)
+	if err != nil {
+		return err
+	}
+	zones, err := ds.ListTrustZones()
+	if err != nil {
+		return fmt.Errorf("failed to list trust zones: %v", err)
+	}
+	table := tablewriter.NewWriter(os.Stdout)
+	table.SetHeader([]string{"Org ID", "TrustZone ID", "Cluster ID", "Trust Zone", "Cluster Name"})
```

Comment:
Perhaps put the cluster ID/name first given this is the resource we're listing?

Might also want to show the profile?

Reply:
I was thinking that if there are many orgs and trust zones, then we'd want them first. Maybe sort/group by those IDs first?

Reply:
Can we check some other CLIs to see what is typical? I feel like the resource identifier generally comes first.

Reply:
I've removed it all.